### PR TITLE
BUGFIX: SD-510 - Export with Active filters context refactor

### PR DESCRIPTION
**Pull request by ptquang86**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist
- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)
- [x] Squash commits

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2304
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2058

# Related Jira tasks
- SD-501: https://jira.intgdc.com/browse/SD-501
- SD-510: https://jira.intgdc.com/browse/SD-510

### CODE REVIEW

**Comment by ptquang86** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1033.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4003/ : SUCCESS in 8m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5071/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1527/ : SUCCESS in 15m 53s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5072/ : SUCCESS in 6s

**Comment by ptquang86** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2179/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1528/ : SUCCESS in 13m 02s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5073/ : SUCCESS in 9s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4006/ : SUCCESS in 8m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5081/ : SUCCESS in 14s

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 1]
```diff
@@ -354,12 +354,37 @@ export function visualizationLoadingHOC<
...
-                const title = exportTitle ? escapeFileName(exportTitle) : "Untitled";
+                const {
+                    format,
+                    includeFilterContext,
+                    mergeHeaders,
+                    showFilters,
+                    title: customTitle,
+                } = exportConfig;
+
+                let title = "Untitled";
```
[the remark this answers is not in the document]
Comment (ptquang86) [reply src/components/core/base/VisualizationLoadingHOC.tsx]: updated :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4008/ : SUCCESS in 6m 03s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5086/ : SUCCESS in 9s

**Comment by nadtan** [comment]
LGTM

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 2]
```diff
@@ -354,12 +354,32 @@ export function visualizationLoadingHOC<
...
+                const title: string = escapeFileName(customTitle || exportTitle) || "Untitled";
+
+                const exportConfigRequest: IExportConfigRequest = {
+                    format,
+                    mergeHeaders,
+                    title,
+                };
+                if (includeFilterContext) {
+                    exportConfigRequest.showFilters = dataSource.getAfm().filters;
+                } else if (showFilters) {
```
Comment (ivan-nejezchleb) [inline src/components/core/base/VisualizationLoadingHOC.tsx]: do we plan to deprecate this direct filters definition for UI SDK components? If yes, there should be some warning/info

File: src/components/core/base/VisualizationLoadingHOC.tsx [hunk 3]
```diff
@@ -354,12 +354,32 @@ export function visualizationLoadingHOC<
...
+                } = exportConfig;
+
+                const title: string = escapeFileName(customTitle || exportTitle) || "Untitled";
+
+                const exportConfigRequest: IExportConfigRequest = {
+                    format,
+                    mergeHeaders,
+                    title,
+                };
+                if (includeFilterContext) {
```
Comment (ivan-nejezchleb) [inline src/components/core/base/VisualizationLoadingHOC.tsx]: Please add definition of this change into changelog and mention priority of these two ways how to define filters for export

File: src/components/core/base/VisualizationLoadingHOC.tsx (hunk 2 above)
Comment (ptquang86) [reply src/components/core/base/VisualizationLoadingHOC.tsx]: removed the backward compatibility's code

File: src/components/core/base/VisualizationLoadingHOC.tsx (hunk 3 above)
Comment (ptquang86) [reply src/components/core/base/VisualizationLoadingHOC.tsx]: removed the backward compatibility's code

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4031/ : SUCCESS in 7m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5121/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4033/ : SUCCESS in 5m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5125/ : SUCCESS in 7s

**Comment by ivan-nejezchleb** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4053/ : SUCCESS in 7m 01s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5152/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2206/ : SUCCESS in 12m 36s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4054/ : SUCCESS in 4m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1056/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1240/ : SUCCESS in 15m 01s

**Pull request merged**